src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -179,6 +181,20 @@ public void run() {
                     }
                 }
             });
+        } else if (SET_APPLICATION_ICON_BADGE_NUMBER.equals(action)) {
```

Comment:
I don't think we need to merge them - the code I wrote handles both cases, because it's called anyway. I'm not sure that handling the same issue from different places in the code is wise